pytorch_toolkit/formula_recognition/tests/test_export.py

```diff
@@ -0,0 +1,121 @@
+
+        def test_2_decoder_export(self):
+            decoder_res_name = self.config.get("res_decoder_name")
+            result_model_exists = os.path.exists(decoder_res_name)
+            if result_model_exists:
+                os.remove(decoder_res_name)
+            self.exporter.export_decoder()
+            result_model_exists = os.path.exists(decoder_res_name)
+            self.assertEqual(True, result_model_exists)
+
+        def test_3_onnx(self):
+            evaluator = Evaluator(self.config, RunnerType.ONNX)
+            metric_onnx = evaluator.validate()
+            target_metric = evaluator.expected_outputs.get("target_metric")
+            self.assertGreaterEqual(metric_onnx, target_metric)
+
+        def test_4_encoder_ir_export(self):
+            if self.config.get("export_ir"):
+                encoder_res_name = self.config.get("res_encoder_name").replace('onnx', 'xml')
+                result_model_exists = os.path.exists(encoder_res_name)
+                if result_model_exists:
+                    os.remove(encoder_res_name)
+                encoder_res_name = self.config.get("res_encoder_name").replace('onnx', 'bin')
+                result_model_exists = os.path.exists(encoder_res_name)
+                if result_model_exists:
+                    os.remove(encoder_res_name)
+
+                self.exporter.export_encoder_ir()
+                encoder_res_name = self.config.get("res_encoder_name").replace('onnx', 'bin')
+                result_model_exists = os.path.exists(encoder_res_name)
+                self.assertEqual(True, result_model_exists)
+                encoder_res_name = self.config.get("res_encoder_name").replace('onnx', 'xml')
+                result_model_exists = os.path.exists(encoder_res_name)
+                self.assertEqual(True, result_model_exists)
+
+        def test_5_decoder_ir_export(self):
+            if self.config.get("export_ir"):
+                decoder_res_name = self.config.get("res_decoder_name").replace('onnx', 'xml')
+                result_model_exists = os.path.exists(decoder_res_name)
+                if result_model_exists:
+                    os.remove(decoder_res_name)
+                decoder_res_name = self.config.get("res_decoder_name").replace('onnx', 'bin')
+                result_model_exists = os.path.exists(decoder_res_name)
+                if result_model_exists:
+                    os.remove(decoder_res_name)
+
+                self.exporter.export_decoder_ir()
+                decoder_res_name = self.config.get("res_decoder_name").replace('onnx', 'bin')
+                result_model_exists = os.path.exists(decoder_res_name)
+                self.assertEqual(True, result_model_exists)
+                decoder_res_name = self.config.get("res_decoder_name").replace('onnx', 'xml')
+                result_model_exists = os.path.exists(decoder_res_name)
+                self.assertEqual(True, result_model_exists)
+
+        def test_6_run_ir_model(self):
```

Comment:
Important question: what if you run `test_6_run_ir_model` as a single test, without running `test_4` and `test_5` before?

As far as I understand, you will receive an error, since the `xml` and `bin` files will be absent.

I think you should call `self.exporter.export_decoder_ir()` and `self.exporter.export_encoder_ir()` beforehand.    
And here there are two options: either call them **always**, or call them if and only if the corresponding `xml+bin` files are absent.    
The first option is safer, the second is faster in terms of tests execution.

Reply:
Added conversion of the model if model does not exists.